### ISSUE TITLE
disable eslint via configuration

### ISSUE THREAD

**Issue by lukeed** [zed-industries/zed#4389]
### Check for existing issues

- [X] Completed

### Describe the feature

It's nice that ESLint support is now added, but I only want it on & running when I want to see it. 

I should be able to disable ESLint via global and project-specific configuration files.
I should _not_ have to create an ESLint config file to disable Zed's internal ESLint ruleset

**Comment by danielweck** [comment]
Same here, would love to be able to completely disabled Zed's built-in ESLINT support until stable/finalised (too many discrepancies with my coding experience in VSCode).
Thanks 👍 :)

**Comment by danielweck** [comment]
Also, `.eslintignore` doesn't seem to have any effect?

**Comment by SomeoneToIgnore** [comment]
To copy my description here: https://github.com/zed-industries/zed/discussions/7204#discussioncomment-8335245

ESLint is yet another LSP server starting for specific languages and currently is very static and not flexible.
All config values it takes are mentioned in the discussions, and local file detection is either done by the corresponding LSP server or not done at all.

**Comment by jerome-benoit** [comment]
It seems to be implemented: 

`settings.json`:

```json
{
  "eslint": {
    "enable": false
  }
}
```

It can even be done on a per directory basis by using `.zed/settings.json`

**Comment by jamaluddinrumi** [comment]
put a blank `eslint.config.mjs` file at the root project directory works for me.

**Comment by danielweck** [comment]
> this is a workaround: put a blank `eslint.config.mjs` file at the root project directory works for me.

That worked for me. Our project uses ESLINT v8 with the old config format anyway, so the Javascript Module "flat" config can live alongside it without interfering with our existing linting setup.

**Comment by danielweck** [comment]
> It seems to be implemented:
> 
> `settings.json`:
> 
> ```json
> {
>   "language_servers": [ "!eslint", "..."]
> }
> ```
> 
> It can even be done on a per directory basis by using `.zed/settings.json`

Yes! https://zed.dev/docs/configuring-languages#choosing-language-servers

**Comment by kaytwo** [comment]
be careful with this snippet at the global settings scope (instead of per-language), as it will implicitly turn on both `vtsls` and `typescript-language-server`: see https://github.com/zed-industries/zed/issues/14544

**Comment by notpeter** [comment]
Given that this is supported:
```json
{
  "language_servers": [ "!eslint", "..."]
}
```
I'm going to go ahead and close.  Thanks for reporting!

**Comment by will-stone** [comment]
Anyone else finding that ESLint server is still starting, even with the recommended `"language_servers": [ "!eslint", "..."]` setting?

**Comment by susyabashti** [comment]
> Anyone else finding that ESLint server is still starting, even with the recommended `"language_servers": [ "!eslint", "..."]` setting?

yeah i do

**Comment by emigdio821** [comment]
> Anyone else finding that ESLint server is still starting, even with the recommended `"language_servers": [ "!eslint", "..."]` setting?

Same here

**Comment by SomeoneToIgnore** [comment]
To mention on https://github.com/zed-industries/zed/issues/4389#issuecomment-2946521208, that got removed, as was breaking more than it fixed: written this way, it would override and re-enable all other `!excluded` servers in all other definitions, e.g. https://github.com/zed-industries/zed/blob/6b2aa1c90aeb72cadfd30ed12141ed9e2569eded/assets/settings/default.json#L2315 would be discarded and `typescript-language-server` re-enabled.

That is bringing the whole set of bugs on itself.

Hence, disabling the esling per language should be done, something like
```
"JavaScript": {
    "language_servers": ["!eslint", "vtsls", "..."]
},
```

**Comment by will-stone** [comment]
@SomeoneToIgnore That did it! Thanks. Here's the full config for those wondering:

```json
{
  "languages": {
    "JavaScript": {
      "language_servers": ["!eslint", "..."]
    },
    "TypeScript": {
      "language_servers": ["!eslint", "..."]
    },
    "TSX": {
      "language_servers": ["!eslint", "..."]
    }
  }
}
```

**Comment by emigdio821** [comment]
It would be nice to set `language_servers` for several languages without keep repeating the same config for each one.
For example:

```json
{
  "languages": {
    "[JavaScript][TypeScript][TSX]": {
      "language_servers": ["!eslint", "..."]
    }
  }
}
```

As far as I remember we can do something like this in VSCode, just as a quick example:

```json
{
  "[typescript][javascript][typescriptreact]": {
    "edior.defaultFormatter": "oxc.oxc-vscode"
  }
}
```
